extensions/objects/templates/ListOfSetsOfHtmlStringsEditor.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+oppia.directive('listOfSetsOfHtmlStringsEditor', [
+  '$compile', 'OBJECT_EDITOR_URL_PREFIX',
+  function($compile, OBJECT_EDITOR_URL_PREFIX) {
+    return {
+      link: function(scope, element) {
+        scope.getTemplateUrl = function() {
+          return OBJECT_EDITOR_URL_PREFIX + 'ListOfSetsOfHtmlStrings';
+        };
+        $compile(element.contents())(scope);
+      },
+      restrict: 'E',
+      scope: {
+        getInitArgs: '&',
+        value: '='
+      },
+      template: '<span ng-include="getTemplateUrl()"></span>',
+      controller: ['$scope', function($scope) {
+        var errorMessage = '';
+
+        if (!$scope.value) {
+          $scope.value = [];
+        }
+        if (!$scope.maxPrevIndex) {
+          $scope.maxPrevIndex = 0;
+        }
+        $scope.initArgs = $scope.getInitArgs();
+        $scope.choices = $scope.initArgs.choices;
+
+        if ($scope.selectedRank !== '' && $scope.selectedRank !== null) {
```

Comment:
What is the initial value of `$scope.selectedRank`?

Reply:
`$scope.selectedRank` contains the value chosen by the learner from the dropdown out of possible ranks. When nothing is selected, '' comes from the form to make its default value. We need not to specify it explicitly.

Reply:
In that case, I would suggest assigning it an initial value so that we know what type to expect and makes that if statement less complicated.

Reply:
Done